src/pages/studies/StudyTableContainer.tsx

```diff
@@ -1,20 +1,48 @@
 import React, { FC } from 'react';
 import { Table } from 'antd';
-import { generateTableData, studiesColumns } from 'store/graphql/studies/models';
+import { generateTableData } from 'store/graphql/studies/models';
 import { StudiesResults } from 'store/graphql/studies/actions';
+import { studiesColumns } from 'store/graphql/studies/tableColumns';
+import { connect, ConnectedProps } from 'react-redux';
+import { createQueryInCohortBuilder, DispatchStoryPage } from 'store/actionCreators/studyPage';
+import { RootState } from 'store/rootState';
+import { Sqon } from 'store/sqon';
 
-const StudyTable: FC<StudiesResults> = (studiesResults) => {
-  const tableData = generateTableData(studiesResults);
+type StudyTableContainerState = {
+  currentVirtualStudy: Sqon[];
+};
+
+const mapDispatch = (dispatch: DispatchStoryPage) => ({
+  onClickStudyLink: (sqons: Sqon[]) => dispatch(createQueryInCohortBuilder(sqons)),
+});
+
+const mapState = (state: RootState): StudyTableContainerState => ({
+  currentVirtualStudy: state.currentVirtualStudy.sqons,
+});
+
+const connector = connect(mapState, mapDispatch);
 
-  if (studiesResults.loading) {
+type PropsFromRedux = ConnectedProps<typeof connector>;
+
+type Props = StudiesResults & PropsFromRedux;
+
+const StudyTable: FC<Props> = (props) => {
```

Comment:
I think it would be easier to read if const {loading, etc} = props. Do we need to pass all the props to the table generator?